core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -119,6 +124,11 @@ public class JDBCUserStoreManager extends AbstractUserStoreManager {
 
     private static final int MAX_ITEM_LIMIT_UNLIMITED = -1;
 
+    // Params added to implement Circuit Breaker
+    private String jdbcConnectionCircuitBreakerState;
+    private long thresholdTimeoutInMilliseconds;
+    private long thresholdStartTime;
```

Comment:
JDBCUserStoreManager.java is actually a singleton as it is registered as an OSGi service. So having class variables like this may affect your logic as there can be multiple writes from multiple threads. So it is better to introduce a ReadWriteLock when reading and writing to these.